pkg/ds/interfaces.go

```diff
@@ -0,0 +1,53 @@
+/*
+Copyright 2023- IBM Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+// Package ds defines and implements a collection of data structures,
+// designed to hold cross-products of two Set structures in a succinct and canonical way.
+// The most important interface is Product. It is implemented by ProductLeft using injective
+// (one-to-one) mapping from sets to sets, where each key-value pair defines a complete cross
+// product of the two sets.
+package ds
+
+type Sized interface {
+	IsEmpty() bool
+	// Size returns the actual, full size of the set. For Product, it returns the number of pairs.
+	Size() int
```

Comment:
should `IsAll()` be also part of the interface?

Reply:
I did not think it is a general property; it requires knowledge about the "universe" which sets may not have.